### PR TITLE
SPCTR-2771 Fix: Global - Fixed an issue wherein a Google error would appear when loading Google Fonts locally with a specific font weight and family set to default.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

- https://brainstormforce.atlassian.net/browse/SPCTR-2771

### Screenshots

- https://share.getcloudapp.com/xQuRXLDn

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

- Added to the card.

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

File: classes/class-uagb-fse-fonts-compatibility.php [hunk 1]
```diff
@@ -201,25 +201,27 @@ public function get_fonts_file_url( $font_family, $font_weight, $font_style ) {
 			}
 			$final_font_files = array();
 
-			foreach ( $font_files[ $font_family_key ] as $key => $font_file ) {
+			if ( is_array( $font_files ) && ! empty( $font_files ) && ! empty( $font_family_key ) ) {
```
Comment (RajkiranBagal) [inline classes/class-uagb-fse-fonts-compatibility.php:204]: If empty or not is_array then just return function, do not add whole foreach in if statement

File: classes/class-uagb-fse-fonts-compatibility.php [hunk 2]
```diff
@@ -466,7 +468,9 @@ public static function delete_all_theme_font_family() {
 
 			// Construct updated theme.json.
 			$theme_json_raw = json_decode( file_get_contents( get_stylesheet_directory() . '/theme.json' ), true );
-
+			if ( ! array_key_exists( 'settings', $theme_json_raw ) || ! array_key_exists( 'typography', $theme_json_raw['settings'] ) || ! array_key_exists( 'fontFamilies', $theme_json_raw['settings']['typography'] ) ) { // Added condition to resolve an issue of PHP Notice:  Undefined index: fontFamilies.
```
Comment (RajkiranBagal) [inline classes/class-uagb-fse-fonts-compatibility.php:471]: Can we just check `! empty( $theme_json_raw['settings] ) ` similar to this for all array_key_exists

File: classes/class-uagb-post-assets.php [hunk 3]
```diff
@@ -622,14 +622,15 @@ public function generate_fonts() {
...
 			}
-			$fonts_attr  .= str_replace( ' ', '+', $gfont_values['fontfamily'] );
-			$fonts_slug[] = sanitize_key( str_replace( ' ', '-', strtolower( $gfont_values['fontfamily'] ) ) );
-
-			if ( ! empty( $gfont_values['fontvariants'] ) ) {
-				$fonts_attr .= ':';
-				$fonts_attr .= implode( ',', $gfont_values['fontvariants'] );
-				foreach ( $gfont_values['fontvariants'] as $key => $font_variants ) {
-					$fonts_attr .= ',' . $font_variants . 'italic';
+			if ( ! empty( $gfont_values['fontfamily'] ) ) {
```
Comment (RajkiranBagal) [inline classes/class-uagb-post-assets.php:625]: Intead of putting whole chunk of code inside if condition, please  write that condition & just continue / break foreach as per requirements.

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: classes/class-uagb-fse-fonts-compatibility.php (hunk 2 above)
Comment (AbhijitNage123) [reply classes/class-uagb-fse-fonts-compatibility.php]: Done and tested

File: classes/class-uagb-fse-fonts-compatibility.php [hunk 4]
```diff
@@ -466,7 +469,9 @@ public static function delete_all_theme_font_family() {
 
 			// Construct updated theme.json.
 			$theme_json_raw = json_decode( file_get_contents( get_stylesheet_directory() . '/theme.json' ), true );
-
+			if ( empty( $theme_json_raw['settings'] ) || empty( $theme_json_raw['settings']['typography'] ) || empty( $theme_json_raw['settings']['typography']['fontFamilies'] ) ) { // Added condition to resolve an issue of PHP Notice:  Undefined index: fontFamilies.
```
Comment (adi3890) [inline classes/class-uagb-fse-fonts-compatibility.php:472]:
```suggestion
			if ( empty( $theme_json_raw['settings']['typography']['fontFamilies'] ) ) { // Added condition to resolve an issue of PHP Notice:  Undefined index: fontFamilies.
```

File: classes/class-uagb-post-assets.php [hunk 5]
```diff
@@ -622,9 +622,11 @@ public function generate_fonts() {
 			if ( ! empty( $fonts_attr ) ) {
 				$fonts_attr .= '|'; // Append a new font to the string.
 			}
+			if ( empty( $gfont_values['fontfamily'] ) ) {
```
Comment (adi3890) [inline classes/class-uagb-post-assets.php:625]: should we also check string check as we  are performing string operations further

File: compatibility/class-uagb-astra-compatibility.php [hunk 6]
```diff
@@ -86,7 +86,7 @@ public function add_google_fonts_in_astra( $astra_fonts ) {
 
 				foreach ( $google_fonts as $key => $gfont_values ) {
 
-					if ( isset( $gfont_values['fontfamily'] ) && isset( $gfont_values['fontvariants'] ) ) {
+					if ( ! empty( $gfont_values['fontfamily'] ) && ! empty( $gfont_values['fontvariants'] ) ) {
```
Comment (adi3890) [inline compatibility/class-uagb-astra-compatibility.php:89]: should we check for expected data type here.

File: compatibility/class-uagb-astra-compatibility.php [hunk 7]
```diff
@@ -86,7 +86,7 @@ public function add_google_fonts_in_astra( $astra_fonts ) {
 
 				foreach ( $google_fonts as $key => $gfont_values ) {
```
Comment (adi3890) [inline compatibility/class-uagb-astra-compatibility.php:87]: @AbhijitNage123  This is quite old function and is very much nested can we add a task in backlog to refactor this function.

File: classes/class-uagb-fse-fonts-compatibility.php (hunk 4 above)
Comment (AbhijitNage123) [reply classes/class-uagb-fse-fonts-compatibility.php]: Resolved

File: classes/class-uagb-post-assets.php (hunk 5 above)
Comment (AbhijitNage123) [reply classes/class-uagb-post-assets.php]: Resolved

File: compatibility/class-uagb-astra-compatibility.php (hunk 6 above)
Comment (AbhijitNage123) [reply compatibility/class-uagb-astra-compatibility.php:92]: Done. If get type for font Family and font Variants then early return

File: compatibility/class-uagb-astra-compatibility.php (hunk 7 above)
Comment (AbhijitNage123) [reply compatibility/class-uagb-astra-compatibility.php:88]: Task link - https://brainstormforce.atlassian.net/browse/SPCTR-2835

**Review by adi3890 — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**